pkg/providers/merge/event.go

```diff
@@ -104,17 +106,61 @@ func (m *merge) ProcessIssueCommentEvent(event *github.IssueCommentEvent) {
 		}
 		util.Error(m.addCanMerge(pr))
 		model := AutoMerge{
-			PrID:      pr.GetNumber(),
-			Owner:     m.owner,
-			Repo:      m.repo,
-			BaseRef:   pr.GetBase().GetRef(),
-			Status:    false,
-			CreatedAt: time.Now(),
+			PrID:         pr.GetNumber(),
+			Owner:        m.owner,
+			Repo:         m.repo,
+			BaseRef:      pr.GetBase().GetRef(),
+			Status:       false,
+			CreatedAt:    time.Now(),
+			WithoutTests: body == withoutTestsMergeCommand,
 		}
+
+		if model.WithoutTests {
+			lastRunAllTestTime, err := m.getLastRunAllTestAt(pr)
+			if err != nil {
+				// When we can not find the last run all tests command, we need to set the without tests flag as false.
+				model.WithoutTests = false
+				util.Error(errors.Wrap(err, "get last run all test time"))
+			} else {
+				model.LastTestAllAt = lastRunAllTestTime
+			}
+		}
+
 		if err := m.saveModel(&model); err != nil {
 			util.Error(errors.Wrap(err, "merge process PR event"))
 		} else {
 			util.Error(m.queueComment(pr))
 		}
 	}
 }
+
+// getLastRunAllTestAt get last run all test comment time.
+func (m *merge) getLastRunAllTestAt(pr *github.PullRequest) (time.Time, error) {
+	if pr == nil {
+		return time.Time{}, errors.New("got a nil pr")
+	}
+
+	var direction = "desc"
+	// Get comments in descending order.
+	options := &github.IssueListCommentsOptions{
+		Direction: &direction,
+	}
```

Comment:
The problem comes here, seems github has removed sort parameters for this API, so we cannot get comments in desc order 😥. You might adjust `perpage` to 100 and list all comments here then find the latest `/run-all-tests` comment.

https://developer.github.com/v3/issues/comments/#list-issue-comments

Reply:
Related issue
https://github.com/google/go-github/issues/1583

Reply:
Got it. I will fix it tonight. Thanks for your support.